crates/lsp/src/main_loop.rs

```diff
@@ -319,14 +389,51 @@ impl GlobalState {
     /// world state could be run concurrently. On the other hand, handlers that
     /// manipulate documents (e.g. formatting or refactoring) should not.
     fn spawn_handler<T, Handler>(
+        &self,
         response_tx: TokioUnboundedSender<anyhow::Result<LspResponse>>,
         handler: Handler,
         into_lsp_response: impl FnOnce(T) -> LspResponse + Send + 'static,
     ) where
         Handler: FnOnce() -> anyhow::Result<T>,
         Handler: Send + 'static,
     {
-        spawn_blocking(move || respond(response_tx, handler(), into_lsp_response).and(Ok(None)))
+        self.spawn_blocking(move || {
+            respond(response_tx, handler(), into_lsp_response).and(Ok(None))
+        })
+    }
+
+    /// Spawn a blocking task
+    ///
+    /// This runs tasks that do semantic analysis on a separate thread pool to avoid
+    /// blocking the main loop.
+    ///
+    /// Can optionally return an event for the auxiliary loop (i.e. a log message or
+    /// diagnostics publication).
+    fn spawn_blocking<Handler>(&self, handler: Handler)
```

Comment:
This is no longer a free function but I think that is fine? Our usage of `spawn_blocking()` in ark is all in functions that get called through `GlobalState`, so we can easily pass around `auxiliary_event_tx` to where we need to call `spawn_blocking()`.

It's possible that `spawn_blocking()` should actually be a method on `AuxiliaryEventSender`, as I think that would end up being cleaner and actually make more sense, since this is a special type of sender.

Reply:
Done in https://github.com/posit-dev/air/pull/80/commits/6407db94b2cea63c9e6c44e7596a70941964cf38